### ISSUE TITLE
Add framework/template for making ICs.

### ISSUE THREAD

**Issue by gbmhunter** [circuitikz/circuitikz#13]
It would be amazing (and I believe feasible) to add a template for helping create ICs. This would be a command which took in the number of pins, the pin number, a brief pin label, name of the IC, and optional pin layout, and created a basic block with the pins arranged around the side, as commonly laid out in programs such as Altium, Eagle, most schematic editors.

**Comment by mredaelli** [comment]
The developer is very lazy, but today tried to get back to business :)

Your idea seems cool, but also kinda challenging

**Comment by der-stefan** [comment]
The idea of drawing ICs is still there. The syntax to add it in circuitikz could look like this:
`` \draw (0,0) node[ic, pins=2x4] (ic1) {};  ``
With the option nxm with n=1...4 and m=1...inf. The size of the box should scale with the number of pins. With this syntax it is not possible to have e.g. 2 pins on the left and 3 pins in the right. Only the same number of pins on every side would be possible.

The pins could be connected by addressing them as follows:
``\draw (ic1.name) node[] {NE555}``
``\draw (ic1.1name) node[right] {VCC}``
``\draw (ic1.1number) node[above] {1}``
``(ic1.1) -- ++(-1,0);``

**Comment by der-stefan** [comment]
It is hard to parse the parameter `pins=2x4`. I think we should split it to `xpins` and `ypins` to avoid the parsing.

**Comment by Rmano** [comment]
You can check http://www.elfsoft2000.com/projects/ --- it's not exactly that, but can give ideas...

**Comment by Rmano** [comment]
Also relevant https://tex.stackexchange.com/questions/146742/creating-anchors-in-a-shape-with-a-foreach-loop

**Comment by Rmano** [comment]
I have done a bit of work on this. This is what I have now; the code: 

```
\begin{circuitikz}[american, ]
    \draw (0,0) node[dipchip,
        hide numbers,
        external pins width=0.3,
        external pad fraction=4 ](A){IC1};
    \foreach \p in {1,...,8}
        \path (A.pin \p) \coord(pin \p) (A.bpin \p) \coordd(bpin \p);
    % second chip
    \draw (A.pin 7) -- ++(2,0) node[dipchip, num pins=14, anchor=pin 3](B){IC2};
    \foreach \p in {center, nw, ne, se, sw, north, south, west, east}
        \path (B.\p) \coord(\p);
    \path (B.text) \coordd(text);
\end{circuitikz}
```

will result in: 

![image](https://user-images.githubusercontent.com/6414907/52291259-43ae0780-2972-11e9-8a03-758b7adad14e.png)

I will prepare a PR with the code, once cleaned up. What do you think? Now supports just DIP (dual in-line package) chips, but should be (almost) straightforward to add quad or similar...

**Comment by Rmano** [comment]
Ok, I think I have finished it. I am open to comments and suggestions, before preparing a PR. 

The code: 

```
\def\coord(#1){coordinate(#1)}
\def\coord(#1){node[circle, red, draw, inner sep=1pt,pin={[red, overlay, inner sep=0.5pt, font=\tiny, pin distance=0.1cm, pin edge={red, overlay,}]45:#1}](#1){}}
\def\coordd(#1){node[circle, red, draw, inner sep=1pt,pin={[red, overlay, inner sep=0.5pt, font=\tiny, pin distance=0.1cm, pin edge={red, overlay,}]-45:#1}](#1){}}

\begin{document}


\begin{circuitikz}[american, ]
    \begin{scope}[yshift=6cm]
        \draw (0,0) node[qfpchip,
            hide numbers,
            num pins = 16,
            external pins width=0.1,
            external pad fraction=4 ](A){IC1};
        \foreach \p in {1,...,16}
            \path (A.pin \p) \coord(pin \p) (A.bpin \p) \coordd(bpin \p);
        \draw (A.pin 9) to[R] ++(2,0) node[qfpchip, num pins=16, anchor=pin 2](B){IC2};
        \draw (B.pin 11) ++(2,0) node[qfpchip, num pins=16, rotate=90]{90}
        ++(3,0) node[qfpchip, num pins=16, rotate=180, rotated numbers]{180}
        ++(-3,-4) node[qfpchip, num pins=16, rotate=-45]{-45};
        \foreach \p in {center, nw, ne, se, sw, north, south, west, east}
            \path (B.\p) \coord(\p);
        \path (B.text) \coordd(text);
    \end{scope}
    \draw (0,0) node[dipchip,
        hide numbers,
        external pins width=0.3,
        external pad fraction=4 ](C){IC1};
    \foreach \p in {1,...,8}
        \path (C.pin \p) \coord(pin \p) (C.bpin \p) \coordd(bpin \p);
    % second chip
        \draw (C.pin 7) to[R] ++(2,0) node[dipchip, num pins=16, anchor=pin 3](D){IC2};
    \foreach \p in {center, nw, ne, se, sw, north, south, west, east}
        \path (D.\p) \coord(\p);
    \path (B.text) \coordd(text);
    \draw (0,-4) node[dipchip, num pins=12, rotate=0]{0};
    \draw (4,-4) node[dipchip, num pins=12, rotate=90, rotated numbers]{90};
    \draw (8,-4) node[dipchip, num pins=12, rotate=50, 
        external pad fraction=4]{50};
    \draw (0,-8) node[dipchip, num pins=12, rotate=180]{180};
    \draw (4,-8) node[dipchip, num pins=12, rotate=-90]{-90};
    \draw (8,-8) node[dipchip, num pins=12, rotate=-40]{-40};
    
\end{circuitikz}
```
will produce the following thing... 

[chips.pdf](https://github.com/circuitikz/circuitikz/files/2841930/chips.pdf)

Small preview: 

![image](https://user-images.githubusercontent.com/6414907/52428615-9c140f00-2b02-11e9-8c11-44168447cc44.png)

**Comment by OmarSquircleArt** [comment]
Nice! I think you got it. Thanks!
